### PR TITLE
Adding image to items

### LINKED ISSUE #17: Add image associated an item
Add the possibility for the user to take a picture of his item and to link it

### PR DESCRIPTION
**Pull request by Arthur-PI**
Close #17 
TODO:
- [x] Add image field in the database
- [x] Allow the user to change the image on the item info page
- [ ] Possibility to add an image at item creation
- [x] Handle no image on item page

### CODE REVIEW

File: supabase/migrations/20231215105229_image-bucket.sql [hunk 1]
```diff
@@ -0,0 +1,45 @@
+drop policy "Anyone can upload an avatar." on "storage"."objects";
+
+drop policy "Avatar images are publicly accessible." on "storage"."objects";
+
```
Comment (victorbillaud) [inline supabase/migrations/20231215105229_image-bucket.sql:4]:
```suggestion
insert into storage.buckets
  (id, name, public)
values
  ('items-images', 'items-images', true);
```

File: utils/lib/item/services.ts
Comment (victorbillaud) [inline utils/lib/item/services.ts:1]: Add tests

File: web/components/common/input/InputFileForm.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          ref={ref}
+          {...props}
+        />
+        <button type='submit' className='hidden' ref={buttonRef} />
+        <Icon
+          name={iconName}
+          size={30}
+          className='absolute left-1/2 top-1/2 z-10 hidden -translate-x-1/2 -translate-y-1/2 group-hover:block'
+          color='text-stone-900 dark:text-white'
+        />
```
Comment (victorbillaud) [inline web/components/common/input/InputFileForm.tsx:39]: Add the loading state using `useFormStatus` cf [SubmitButton](https://github.com/victorbillaud/tomato/blob/d9b23e7b656e41d5cb8f33d75f1cfca9ab4b0505/web/components/common/button/SubmitButton.tsx)

**Review by victorbillaud — CHANGES_REQUESTED** [review]

File: utils/lib/item/services.ts
Comment (Arthur-PI) [reply utils/lib/item/services.ts:1]: done

File: web/components/common/input/InputFileForm.tsx (hunk 2 above)
Comment (Arthur-PI) [reply web/components/common/input/InputFileForm.tsx:59]: Done

**Review by victorbillaud — APPROVED** [review]

**Pull request merged**